src/ShellRouting/IShellRouter.cs

```diff
@@ -58,7 +69,41 @@ public interface IShellRouter
         ///         );
         ///     ]]></code>
         /// </example>
-        public void RegisterHandler(string messageType, Action<Message> handler);
+        public void RegisterHandler(string messageType, Action<Message> handler)
```

Comment:
as before, I would only keep the async version and "force" all handler implementers to "think asynchronously" :-)